src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java

```diff
@@ -190,21 +190,32 @@ private Network networksListToMergedNetwork(List<Network> networks) {
         }
     }
 
-    ExportNetworkInfos exportNetwork(UUID networkUuid, List<UUID> otherNetworksUuid, String format) throws IOException {
+    ExportNetworkInfos exportNetwork(UUID networkUuid, String variantId, List<UUID> otherNetworksUuid, String format) throws IOException {
         if (!Exporters.getFormats().contains(format)) {
             throw NetworkConversionException.createFormatUnsupported(format);
         }
         MemDataSource memDataSource = new MemDataSource();
 
         Network network = networksListToMergedNetwork(getNetworkAsList(networkUuid, otherNetworksUuid));
+        if (variantId != null) {
+            if (network.getVariantManager().getVariantIds().contains(variantId)) {
+                network.getVariantManager().setWorkingVariant(variantId);
+            } else {
+                throw NetworkConversionException.createVariantIdUnknown(variantId);
+            }
+        }
 
         Exporters.export(format, network, null, memDataSource);
 
         Set<String> listNames = memDataSource.listNames(".*");
         String networkName;
         byte[] networkData;
         if (listNames.size() == 1) {
-            networkName = network.getNameOrId() + listNames.toArray()[0];
+            networkName = network.getNameOrId();
+            if (variantId != null) {
```

Comment:
Maybe always add variant id in the file name ?
If variantId is null, you can use VariantManagerConstants.INITIAL_VARIANT_D

Reply:
Done

Reply:
The user does not know the notion of variant ?

Reply:
True. Checked with Christine. We should add a user defined name for exported file later.
For now and this scope it's ok.

Reply:
ok